app/Filament/Resources/ApiKeyResource/Pages/ListApiKeys.php

```diff
@@ -51,13 +52,23 @@ public function table(Table $table): Table
             ])
             ->actions([
                 DeleteAction::make(),
+            ])
+            ->emptyStateIcon('tabler-key')
+            ->emptyStateDescription('')
+            ->emptyStateHeading('No API Keys')
+            ->emptyStateActions([
+                CreateAction::make('create')
+                    ->label('Create API Key')
+                    ->button(),
             ]);
     }
 
     protected function getHeaderActions(): array
     {
         return [
-            Actions\CreateAction::make(),
+            Actions\CreateAction::make()
+                ->label('New API Key')
+                ->hidden(fn () => ApiKey::count() <= 0),
```

Comment:
![image](https://github.com/user-attachments/assets/b6ee5abf-6bfa-4f56-aa1f-c4c6f5555c7c)
You are checking for all API keys, this also includes user API keys, resulting in both buttons appearing if a client has an API key, we need to change for `key_type`  = `2` 

1 = User
2 = Application